Jenkinsfile

```diff
@@ -18,315 +18,326 @@ pipeline {
-					CONTAINER_HOST = getContainerHostName('NRESPipelineTest-NRESPipelineTest-1')
-				}
-			}
+                        podName = sh(script: 'kubectl -n dev get po -l app.kubernetes.io/instance=nres-pipe ' +
+                                        '--sort-by=.status.startTime -o jsonpath="{.items[-1].metadata.name}"',
+                                     returnStdout: true).trim()
+
+                    }
+                 }
+		    }
 		}
-		stage('Test-Bias-Ingestion') {
+		stage("Test-Bias-Ingestion") {
 			when {
 				anyOf {
-					branch 'PR-*'
+					branch "PR-*"
 					expression { return params.forceEndToEnd }
 				}
 			}
 			steps {
 				script {
-					sshagent(credentials: ['jenkins-rancher-ssh']) {
-						executeOnRancher('pytest --durations=0 --junitxml=/nres/code/pytest-bias-ingestion.xml ' +
-								'-m bias_ingestion /nres/code/',
-						    CONTAINER_HOST, CONTAINER_ID, ARCHIVE_UID)
+                    withKubeConfig([credentialsId: "prod-kube-config"]) {
+						sh("kubectl -n dev exec -c nres-pipeline ${podName} -- " +
+						        "/usr/bin/sudo -s -u archive pytest --durations=0 " +
+						        "--junitxml=/nres/code/pytest-bias-ingestion.xml -m bias_ingestion /nres/code/")
 					}
 				}
 			}
 			post {
 				always {
-					script{
-						sshagent(credentials: ['jenkins-rancher-ssh']) {
-							copyFromRancherContainer('/nres/code/pytest-bias-ingestion.xml', '.', CONTAINER_HOST, CONTAINER_ID)
+					script {
+					    withKubeConfig([credentialsId: "prod-kube-config"]) {
+						    sh("kubectl -n dev cp -c nres-pipeline " +
+						            "${podName}:/nres/code/pytest-bias-ingestion.xml " +
+						            "pytest-bias-ingestion.xml")
+						    junit "pytest-bias-ingestion.xml"
 						}
-						junit 'pytest-bias-ingestion.xml'
 					}
 				}
 			}
 		}
-		stage('Test-Master-Bias-Creation') {
+		stage("Test-Master-Bias-Creation") {
 			when {
 				anyOf {
-					branch 'PR-*'
+					branch "PR-*"
 					expression { return params.forceEndToEnd }
 				}
 			}
 			steps {
 				script {
-					sshagent(credentials: ['jenkins-rancher-ssh']) {
-						executeOnRancher('pytest --durations=0 --junitxml=/nres/code/pytest-master-bias.xml ' +
-								'-m master_bias /nres/code/',
-								CONTAINER_HOST, CONTAINER_ID, ARCHIVE_UID)
-					}
-				}
+			        withKubeConfig([credentialsId: "prod-kube-config"]) {
+                        sh("kubectl -n dev exec -c nres-pipeline ${podName} -- " +
+                                "/usr/bin/sudo -s -u archive pytest --durations=0 " +
+                                "--junitxml=/nres/code/pytest-master-bias.xml -m master_bias /nres/code/")
+			        }
+			    }
 			}
 			post {
 				always {
-					script{
-						sshagent(credentials: ['jenkins-rancher-ssh']) {
-							copyFromRancherContainer('/nres/code/pytest-master-bias.xml', '.', CONTAINER_HOST, CONTAINER_ID)
+					script {
+					    withKubeConfig([credentialsId: "prod-kube-config"]) {
```

Comment:
If all these steps use the `dev` namespace, you can use the `dev-kube-config` credentials.

Reply:
Fixed.